src/main/osd/osd.c

```diff
@@ -1358,7 +1362,10 @@ bool osdUpdateCheck(timeUs_t currentTimeUs, timeDelta_t currentDeltaTimeUs)
 
             // Determine time of next update
             if (osdUpdateDueUs) {
-                osdUpdateDueUs += OSD_UPDATE_INTERVAL_US;
+                // Ensure there's not a flurry of updates to catch up
+                while (osdUpdateDueUs < currentTimeUs) {
```

Comment:
```suggestion
                while (cmp32(osdUpdateDueUs, currentTimeUs) < 0) {
```

This code is also a bit fragile - you can end up counting to 2^31 when osdUpdateDueUs  is invalid

Reply:
Thanks for that. I normally use that automatically...